src/Symfony/Component/PropertyInfo/Extractor/ReflectionExtractor.php

```diff
@@ -384,11 +384,11 @@ public function getReadInfo(string $class, string $property, array $context = []
         }
 
         if ($allowMagicGet && $reflClass->hasMethod('__get') && (($r = $reflClass->getMethod('__get'))->getModifiers() & $this->methodReflectionFlags)) {
-            return new PropertyReadInfo(PropertyReadInfo::TYPE_PROPERTY, $property, PropertyReadInfo::VISIBILITY_PUBLIC, false, $r->returnsReference());
+            return new PropertyReadInfo(PropertyReadInfo::TYPE_PROPERTY, $property, PropertyReadInfo::VISIBILITY_PUBLIC, false, $r->returnsReference(), false, false);
```

Comment:
Why don't we use `propertyHasHook` and `propertyIsVirtual` here?

Reply:
Good catch, I swapped the `$allowMagicGet` and the `$hasProperty` cases instead, so we have the same priorities for read & write and we know in the `$allowMagicGet` case that there is no property.